detectors/node/opentelemetry-resource-detector-aws/src/detectors/AwsEcsDetector.ts

```diff
@@ -77,13 +91,137 @@ export class AwsEcsDetector implements Detector {
       const splitData = rawData.trim().split('\n');
       for (const str of splitData) {
         if (str.length > this.CONTAINER_ID_LENGTH) {
-          return str.substring(str.length - this.CONTAINER_ID_LENGTH);
+          containerId = str.substring(str.length - this.CONTAINER_ID_LENGTH);
+          break;
         }
       }
     } catch (e) {
-      diag.warn(`AwsEcsDetector failed to read container ID: ${e.message}`);
+      diag.warn('AwsEcsDetector failed to read container ID', e);
+    }
+
+    if (hostName || containerId) {
+      return new Resource({
+        [SemanticResourceAttributes.CONTAINER_NAME]: hostName || '',
```

Comment:
Honestly, I do not know. All ECS detectors I found across the various SDKs that implement them, came with this behavior before I set out to add to them. It seems to me that ECS seems to set the container name as hostname rather consistently, both in EC2 and Fargate. It could be overridden with something more precise if the metadata endpoint is found (the container name comes in with that too) by parsing the container ARN, but it is not something I would do in this PR.